src/layout/TagsInput.js

```diff
@@ -79,87 +81,109 @@ let TagsInput = withState('state', 'setState', () =>
       PopoverContents,
       style,
     }) => {
+      let containerRef
+      let inputRef
       if (splitCommas)
         addTag = _.flow(
           _.split(','),
           _.map(addTag)
         )
       return (
-        <div className="tags-input" style={{ ...style }}>
-          <Flex
-            style={{
-              cursor: 'text',
-              alignItems: 'center',
-              flexWrap: 'wrap',
-              height: '100%',
-              padding: 3,
+        <OutsideClickHandler
+          onOutsideClick={() => {
+            state.isOneLine = true
+            containerRef.scrollTop = 0
+          }}
+        >
+          <div
+            className={`tags-input ${
+              state.isOneLine ? 'tags-input-one-line' : ''
+            }`}
+            ref={e => (containerRef = e ? e : containerRef)}
+            style={{ ...style }}
+            onClick={() => {
+              if (state.isOneLine) {
+                state.isOneLine = false
+                inputRef.focus()
+              }
             }}
           >
-            {_.map(
-              t => (
-                <TagComponent
-                  key={t}
-                  value={t}
-                  {...{ removeTag, tagStyle }}
-                  onClick={() => {
-                    state.popoverOpen = true
-                    state.selectedTag = t
-                  }}
-                />
-              ),
-              tags
-            )}
-            <input
+            <Flex
               style={{
-                border: 'none',
-                outline: 'none',
-                flex: 1,
-                margin: 3,
-                minWidth: 120,
-              }}
-              onChange={e => {
-                state.currentInput = e.target.value
+                cursor: 'text',
+                alignItems: 'center',
+                flexWrap: 'wrap',
+                height: '100%',
+                padding: 2,
```

Comment:
FYI, the `alignItems` and `flexWrap` properties are now part of our Flex component API, so you could also do:

```jsx
<Flex 
  wrap
  alignItems="center"
  style={{
    cursor: 'text',
    height: '100%',
    padding: 2 
  }}
>
```

Reply:
Sure it seems to work so why not. Changes made.